hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/SCMCommonPlacementPolicy.java

```diff
@@ -126,7 +127,36 @@ public ConfigurationSource getConf() {
    * @throws SCMException SCM exception.
    */
   @Override
-  public List<DatanodeDetails> chooseDatanodes(
+  public final List<DatanodeDetails> chooseDatanodes(
+          List<DatanodeDetails> excludedNodes,
+          List<DatanodeDetails> favoredNodes,
+          int nodesRequired, long metadataSizeRequired, long dataSizeRequired)
+          throws SCMException {
+//    This method calls the chooseDatanodeInternal after fixing
```

Comment:
Expand this comment a bit, and also mention this Jira in it please. Eg:

> The nodes in the exclude list are created from the ExcludeList passed from the client. It contains only the DN UUID to exclude and does not include the network location. In order for the node choosing code to work correctly, it needs to know the network location (rack) the node is associated with. Therefore we lookup each excluded node in the NodeManager by its UUID, and if found, use the datanodeDetails instance stored inside NodeManager. If no node is found in NM, then it means SCM does know know about this node, so there is no point in passing it into the downstream methods and it can be filtered out. See HDDS-7015.

Reply:
Done